### PR TITLE
Added unit tests for Gauge and counter metrics

### PR DESCRIPTION
**Pull request by Manoramsharma**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

Added unit tests for `Gauge` and Counter `metrics` under `pkg/kwok/metrics`

### CODE REVIEW

File: pkg/kwok/metrics/counter_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2023 The Kubernetes Authors.
```
Comment (wzshiming) [inline pkg/kwok/metrics/counter_test.go:2]:
```suggestion
Copyright 2024 The Kubernetes Authors.
```

File: pkg/kwok/metrics/gauge_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2023 The Kubernetes Authors.
```
Comment (wzshiming) [inline pkg/kwok/metrics/gauge_test.go:2]:
```suggestion
Copyright 2024 The Kubernetes Authors.
```

File: pkg/kwok/metrics/counter_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	initialValue := float64(0)
+
+	// Create a new counter and ensure its atomic pointer is initialized properly
+	c := &counter{
+		value: &atomic.Pointer[float64]{},
+	}
+	c.value.Store(&initialValue)
+	c.CounterFunc = prometheus.NewCounterFunc(opts, func() float64 {
+		return *c.value.Load()
+	})
```
Comment (wzshiming) [inline pkg/kwok/metrics/counter_test.go:45]:
```suggestion
	// Create a new counter
	c := NewCounter(opts)
```

File: pkg/kwok/metrics/counter_test.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+	// Register the counter with Prometheus's default registry
+	prometheus.MustRegister(c)
```
Comment (wzshiming) [inline pkg/kwok/metrics/counter_test.go:48]:
```suggestion
```

File: pkg/kwok/metrics/counter_test.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	if err := testutil.CollectAndCompare(c, strings.NewReader(`
+		# HELP test_counter This is a test counter
+		# TYPE test_counter counter
+		test_counter 84
+	`)); err != nil {
+		t.Fatalf("unexpected metrics after second set: %s", err)
+	}
+
+	// Clean up the counter from the default registry
+	prometheus.Unregister(c)
```
Comment (wzshiming) [inline pkg/kwok/metrics/counter_test.go:82]:
```suggestion
```

File: pkg/kwok/metrics/gauge_test.go [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+	opts := GaugeOpts{
+		Name: "test_gauge",
+		Help: "This is a test gauge",
+	}
+
+	// Create a new gauge
+	gauge := NewGauge(opts)
+
+	// Register the gauge with Prometheus's default registry
+	prometheus.MustRegister(gauge)
```
Comment (wzshiming) [inline pkg/kwok/metrics/gauge_test.go:38]:
```suggestion
```

File: pkg/kwok/metrics/gauge_test.go [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+	if err := testutil.CollectAndCompare(gauge, strings.NewReader(`
+		# HELP test_gauge This is a test gauge
+		# TYPE test_gauge gauge
+		test_gauge 42
+	`)); err != nil {
+		t.Fatalf("unexpected metrics after set: %s", err)
+	}
+
+	// Clean up the gauge from the default registry
+	prometheus.Unregister(gauge)
```
Comment (wzshiming) [inline pkg/kwok/metrics/gauge_test.go:80]:
```suggestion
```

File: pkg/kwok/metrics/counter_test.go (hunk 3 above)
Comment (wzshiming) [reply pkg/kwok/metrics/counter_test.go:45]: initialize the counter in NewCounter
`c.value.Store(new(float64))`

**Comment by Manoramsharma** [comment]
Hi @wzshiming how can I fix this failing test particularly for binary run time, with workflow error.                                                 ` Test kwokctl/kwokctl_binary_benchmark failed.`

**Review by wzshiming — COMMENTED** [review]
/approve
/lgtm
/label tide/merge-method-squash

**Pull request merged**